addons/gamification/models/gamification_challenge.py

```diff
@@ -302,95 +283,64 @@ def _update_all(self, cr, uid, ids, context=None):
+        self._recompute_challenge_users()
+        self._generate_goals_from_challenge()
 
-            if challenge.last_report_date != fields.date.today():
+        for challenge in self:
+            if challenge.last_report_date != fields.Date.today():
                 # goals closed but still opened at the last report date
-                closed_goals_to_report = goal_obj.search(cr, uid, [
+                closed_goals_to_report = Goal.search([
                     ('challenge_id', '=', challenge.id),
                     ('start_date', '>=', challenge.last_report_date),
                     ('end_date', '<=', challenge.last_report_date)
                 ])
 
-                if challenge.next_report_date and fields.date.today() >= challenge.next_report_date:
-                    self.report_progress(cr, uid, challenge, context=context)
+                if challenge.next_report_date and fields.Date.from_string(fields.Date.today()) >= fields.Date.from_string(challenge.next_report_date):
+                    challenge.report_progress()
 
-                elif len(closed_goals_to_report) > 0:
+                elif len(closed_goals_to_report.ids) > 0:
                     # some goals need a final report
-                    self.report_progress(cr, uid, challenge, subset_goal_ids=closed_goals_to_report, context=context)
+                    challenge.report_progress(subset_goal_ids=closed_goals_to_report.ids)
 
-        self.check_challenge_reward(cr, uid, ids, context=context)
+        self.check_challenge_reward()
         return True
 
-    def quick_update(self, cr, uid, challenge_id, context=None):
-        """Update all the goals of a specific challenge, no generation of new goals"""
-        goal_ids = self.pool.get('gamification.goal').search(cr, uid, [('challenge_id', '=', challenge_id)], context=context)
-        self.pool.get('gamification.goal').update_goal(cr, uid, goal_ids, context=context)
-        return True
-
-    def _get_challenger_users(self, cr, uid, domain, context=None):
-        user_domain = eval(ustr(domain))
-        return self.pool['res.users'].search(cr, uid, user_domain, context=context)
+    def _get_challenger_users(self, domain):
+        return self.env['res.users'].search(eval(ustr(domain))).ids
 
-    def _recompute_challenge_users(self, cr, uid, challenge_ids, context=None):
+    def _recompute_challenge_users(self):
         """Recompute the domain to add new users and remove the one no longer matching the domain"""
-        for challenge in self.browse(cr, uid, challenge_ids, context=context):
+        for challenge in self:
             if challenge.user_domain:
 
-                old_user_ids = [user.id for user in challenge.user_ids]
-                new_user_ids = self._get_challenger_users(cr, uid, challenge.user_domain, context=context)
+                old_user_ids = challenge.user_ids.ids
+                new_user_ids = self._get_challenger_users(challenge.user_domain)
```

Comment:
You can make _get_challenger_users method returns recordset and at line 323, 324 instead of explicitly performing set operation, you can use over recordset